server/src/main/java/org/elasticsearch/common/xcontent/LoggingDeprecationHandler.java

```diff
@@ -24,39 +25,60 @@
  * though the user sent them.
  */
 public class LoggingDeprecationHandler implements DeprecationHandler {
-    public static final LoggingDeprecationHandler INSTANCE = new LoggingDeprecationHandler();
     /**
      * The logger to which to send deprecation messages.
-     *
+     * <p>
      * This uses ParseField's logger because that is the logger that
      * we have been using for many releases for deprecated fields.
      * Changing that will require some research to make super duper
      * sure it is safe.
      */
     private static final DeprecationLogger deprecationLogger = DeprecationLogger.getLogger(ParseField.class);
 
-    private LoggingDeprecationHandler() {
-        // Singleton
+    public static final LoggingDeprecationHandler INSTANCE = new LoggingDeprecationHandler((message, params, field_name) ->
```

Comment:
Can you make these singleton instances private, so the only access is via the getter method ?

Reply:
`LoggingDeprecationHandler.INSTANCE` is used in 241 places. I am afraid that this would hide the main intent of this PR.

Reply:
also please note that this getter method is actually an instance method..